### ISSUE TITLE
check all markdown files recursively in pre-commit workflow

### ISSUE THREAD

**Issue by hamkuu** [igorshubovych/markdownlint-cli#461]
## Issue

I can check all markdown files recursively by running

```bash
markdownlint **/*.md
```

But I haven't make it happen via pre-commit workflow.

The sample provided on [#use-with-pre-commit](https://github.com/igorshubovych/markdownlint-cli?tab=readme-ov-file#use-with-pre-commit) only checks markdown files on the root directory:

```bash
- repo: https://github.com/igorshubovych/markdownlint-cli
  rev: v0.39.0
  hooks:
  - id: markdownlint
```

## Attempted Solution

By adding `files` config as follow, it still doesn't work as expected:

```bash
- repo: https://github.com/igorshubovych/markdownlint-cli
  rev: v0.39.0
  hooks:
  - id: markdownlint
    files: '\.md$'
```

## Question

May I know how I can check all files recursively in pre-commit configurations?

**Comment by DavidAnson** [comment]
This is a pre-commit question and I don't use that tool. However, it looks like `files` is meant to take a regular expression and I'm not sure how you would tell it to include subdirectories. Instead, you should be able to use the normal command-line syntax with `args` and `**/*.md` instead.

**Comment by hamkuu** [comment]
> This is a pre-commit question and I don't use that tool. However, it looks like `files` is meant to take a regular expression and I'm not sure how you would tell it to include subdirectories. Instead, you should be able to use the normal command-line syntax with `args` and `**/*.md` instead.

Thanks! I can confirm `args` works as expected:

```yaml
  - repo: https://github.com/igorshubovych/markdownlint-cli
    rev: v0.39.0
    hooks:
      - id: markdownlint
        args:
          - '**/*.md'
```
